src/include/uapi/abi.h

```diff
@@ -51,8 +51,8 @@
 #define __INCLUDE_UAPI_ABI_H__
 
 /** \brief SOF ABI version major, minor and patch numbers */
-#define SOF_ABI_MAJOR 3
-#define SOF_ABI_MINOR 6
+#define SOF_ABI_MAJOR 4
```

Comment:
I can't see where the ABI is changed ? Is there a patch missing from the PR ?

Reply:
@lgirdwood there are no changes to the IPC structures in this PR but I changed the ABI because the way the link DMA channels are set is different. So kernel that sets the link DMA channel dynamically would expect the FW to do the same and if the FW is older where the link DMA channels are set statically, all HDA usecases would fail. So they are incompatible. I didnt know how else to call out this mismatch.

Reply:
ok, but I also dont see any changes to the IPC programming flow (including in the kernel PR), this looks more like a dependency on register programming flow depending on how FW allocates HDA DMA channels ? If so, it's probably better to advertise this (using boot extended metadata) so the driver can then decide which programming version to use. This also means that 1:N could still be supported in FW/driver (and re-used when HW supports it).

Reply:
@lgirdwood I could advertise this with boot extended metadata but I don't understand how this would help the 1:N case. This scenario would be broken even with the static allocation scheme that we have today.

Reply:
@ranj063 ok, so you mean that "change will not work with 1:N FE <-> BE connections" is even true today without this change.

Reply:
@lgirdwood yes thats correct

Reply:
@lgirdwood the boot ext metadata would handle the backward compatibility between a new kernel and an older FW. But do you have any suggestions for newer FW and older kernel? 
We send the link_dma_ch as part of the DAI config IPC. In the case of dynamic allocation, this would still be there but just that we ignore the link_dma_ch field which will always be 0 from the driver.

Reply:
@lgirdwood answering my question myself. I think I've found a way to do this now. I will update the PR soon